server/proto_server.go

```diff
@@ -0,0 +1,52 @@
+package server
+
+import (
+	"net"
+	"fmt"
+	"google.golang.org/grpc"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+func startRPCServer(port int) error {
+	lis, err := net.Listen("tcp", fmt.Sprintf("%s:%d", "localhost", port - 1))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+	s := Server{}
+	grpcServer := grpc.NewServer()
+	RegisterClientServer(grpcServer, &s)
+	go func() {
+		log.Fatal(grpcServer.Serve(lis))
+	}()
+
+
+	go func() error {
+		err := startRESTServer(port)
+		if err != nil {
+			log.Fatalf("failed to start gRPC REST server: %s", err)
+		}
+		return err
+	}()
+
+	return err
+}
+
+func startRESTServer(port int) error {
```

Comment:
Is this a RESTful server or simply a version of the gRPC service that supports JSON?

Reply:
Its a version of gRPC service that supports RESTful JSON API into gRPC